ext/rbs_extension/main.c

```diff
@@ -51,6 +51,7 @@ static void declare_type_variables(parserstate *parser, VALUE variables, VALUE b
   if (NIL_P(variables)) return; // Nothing to do.
 
   if (!RB_TYPE_P(variables, T_ARRAY)) {
+    free_parser(parser);
     rb_raise(rb_eTypeError,
```

Comment:
Hmmm, I wonder if we can leverage the existing `ensure` blocks to centralize the freeing there:

https://github.com/Shopify/rbs/blob/93a23bd8b342f0b2773b9a2c91ce4e69a46e3cf4/ext/rbs_extension/main.c#L179

That would take some rearranging (to move lines 170-171 into the ensured-function `parse_type_try()`)... 🤔 

I would ask for Peter's advice on how to proceed.

Reply:
(This isn't blocking, though)